feature/se-43

j0world opened the pull request:
close [SE-WS23/documentation#43 ](https://github.com/SE2-WS23/documentation/issues/43)

![86BC4C00-ACCF-4AEE-B067-36A97DB7594E](https://github.com/SE2-WS23/Skillverse-Web/assets/136974808/dc578372-51fe-470d-82b9-d7e864cde715)
implements the quiz page 
currently works on dummy data, next button doesn't do anything yet

j0world committed:
Add: quiz page
added quiz page and modified main.jsx to include a link to it

julix14 commented on src/main.jsx:
```diff
@@ -46,6 +47,11 @@ const router = createBrowserRouter([
     element: <AssessmentPage />,
     errorElement: <ErrorPage />,
   },
+  {
+    path: "/quiz",
+    element: <QuizPage />,
+    errorElement: <ErrorPage />,
+  },
```
Please merge the current main branch into yours, there was a bigger change in the router and i think you have to adapt your code to it

julix14 commented on src/pages/Quiz/QuizPage.jsx:
```diff
@@ -0,0 +1,120 @@
+                flexDirection: "column",
+                justifyContent: "center",
+                alignItems: "center",
+                height: "calc(100% - 60px)",
```
Why did you calculate the hight here?

j0world replied:
Honestly, idk. Copied from assessment page for uniformity

julix14 commented on src/pages/Quiz/QuizPage.jsx:
```diff
@@ -0,0 +1,120 @@
+
+function QuizPage() {
+
+    const [currentQuestion, setcurrentQuestion] = useState(0);
+    const [progressValue, setprogressValue] = useState(1);
```
```suggestion
    const [currentQuestion, setCurrentQuestion] = useState(0);
    const [progressValue, setProgressValue] = useState(1);
```

j0world replied:
done

julix14 reviewed (changes requested):
<img width="1378" alt="image" src="https://github.com/SE2-WS23/Skillverse-Web/assets/65715000/c19dcc5b-9889-4b61-b8f8-117eba6172c7">
I think you messed up the layout a bit. If you change the screenheight a bit it looks like this

Chrissydrx commented on src/pages/Quiz/QuizPage.jsx:
Just as a general question: Should we implement an implementation if a question is answered correct or wrong?

Chrissydrx commented on src/pages/Quiz/QuizPage.jsx:
```diff
@@ -0,0 +1,120 @@
+function QuizPage() {
+
+    const [currentQuestion, setcurrentQuestion] = useState(0);
+    const [progressValue, setprogressValue] = useState(1);
```
Why is the default value for the progress Bar 1?

j0world replied:
It was the value I had in there when it finally started working. It also sets a minimal amount of progress even before answering the first question and I thought that looked cuter :)

Chrissydrx replied:
Why does it not work with 0?

j0world replied:
it does work with zero, I just think it looks better with one

Chrissydrx commented on src/pages/Quiz/QuizPage.jsx:
```diff
@@ -0,0 +1,120 @@
+        else
+            console.log("Incorrect");
+
+        if (currentQuestion !== quiztions.length - 1) { // updates questions and progress bar
+            setcurrentQuestion(currentQuestion + 1);
+            setprogressValue((((currentQuestion +1)/ quiztions.length) * 100));
+        }
```
```suggestion
        if (currentQuestion !== quiztions.length - 1) { // updates questions and progress bar
            setcurrentQuestion(currentQuestion + 1);
            setprogressValue((((currentQuestion +1)/ quiztions.length) * 100));
            return;
        }
```

j0world replied:
done

Chrissydrx commented on src/pages/Quiz/QuizPage.jsx:
```diff
@@ -0,0 +1,120 @@
+        else
+            console.log("Incorrect");
+
+        if (currentQuestion !== quiztions.length - 1) { // updates questions and progress bar
```
The comment is miss leading and is not necessary

j0world replied:
deleted

Chrissydrx commented on src/pages/Quiz/QuizPage.jsx:
```diff
@@ -0,0 +1,120 @@
+            setprogressValue(100); 
+    }
+
+    const quiztions = [ // placeholder for the real questions
```
What are quiztions? Do you mean questions?
Maybe you should name it dummyQuestions

j0world replied:
I thought it was a funny wordplay. It is changed now

Chrissydrx replied:
Please don't use wordplays in coding. You variables should be self-explanatory and nobody should guess what you mean with it.

Chrissydrx commented on src/pages/Quiz/QuizPage.jsx:
```diff
@@ -0,0 +1,120 @@
+            >
+                <LinearProgress 
+                    sx={{
+                    margin: "4% 0 3% 0",
```
Margin in percent? I think, this can lead to unwanted effects

j0world replied:
yep, I just learned that vw and vh are a thing :)
please check if it is better now

Chrissydrx commented on src/pages/Quiz/QuizPage.jsx:
```diff
@@ -0,0 +1,120 @@
+                    sx={{
+                    height: "80%",
+                    width: "80%",
+                    backgroundColor: "#D9D9D9",
```
Please you theme data for colors

j0world replied:
I don't quite understand. The grey tone is the same as in the Refined Wireframe and the assessment page. Should I import the theme from somewhere?

Chrissydrx replied:
The assessment page is right now only a placeholder. So, it is not really part of the theme. The Background color of the Quiz will be final and should reflect the theme. So, please you theme data

j0world replied:
The background color is just the same as in the wire frame and I thought that was fine. 
As I still don't quite understand, would you mind rephrasing what I need to do?
Is the grey box also supposed to turn Blue like the buttons?

j0world replied:
![Screenshot 2023-11-13 at 19 51 26](https://github.com/SE2-WS23/Skillverse-Web/assets/136974808/5259284f-1c61-4ef8-a4e9-1c04447a7c3a)
:(

I have yet to push this and would like clarification whether this is what you meant.

j0world replied:
![Screenshot 2023-11-15 at 15 25 31](https://github.com/SE2-WS23/Skillverse-Web/assets/136974808/9d512e0d-1e5c-4093-a75c-edf0c3649621)


Quiz with the new light blue, please say if this is ok

Chrissydrx commented on src/pages/Quiz/QuizPage.jsx:
```diff
@@ -0,0 +1,120 @@
+                    }}
+                >
+                    
+                    <h1 style={{margin: "0% 0 15% 0"}}>
```
Margin in percent? I think, this can lead to unwanted effects

j0world replied:
changed to vh/ vw

Chrissydrx commented on src/pages/Quiz/QuizPage.jsx:
```diff
@@ -0,0 +1,120 @@
+                            variant="contained"
+                            color="primary"
+                            size="large"
+                            sx={{ width: "70%", margin: "2% 0"}}
```
Margin in percent? I think, this can lead to unwanted effects

j0world replied:
changed to vh/ vw

Chrissydrx commented on src/pages/Quiz/QuizPage.jsx:
```diff
@@ -0,0 +1,120 @@
+                flexDirection: "column",
+                justifyContent: "center",
+                alignItems: "center",
+                height: "calc(100% - 60px)",
```
The height overlaps with the button

Chrissydrx commented on src/pages/Quiz/QuizPage.jsx:
<img width="949" alt="image" src="https://github.com/SE2-WS23/Skillverse-Web/assets/37250473/27e76e79-a867-4c63-8061-cf6ee38b6dd6">
The page is not responsive

j0world replied:
hopefully fixed, it's better now for sure

j0world committed:
Merge branch 'main' into feature/se-43

j0world committed:
fix: implementing recommended changes
making layout more reactive (change from % to vw/vh)
setcurrentQuestion -> setCurrentQuestion
setcurrentAnswer -> setCurrentAnswer
added return to handleAnswerButtonClick()
deleted unnecessary comment handleAnswerButtonClick()
quiztions -> dummyQuestions

j0world committed:
fix: re added LinkedInLearningPage
I think it came here after the merge?
didn't think I'd also have to commit it but here we are :)

j0world committed:
Merge branch 'main' into feature/se-43

j0world committed:
fix: re added quiz to the router

j0world committed:
fix: added theme data colors
also added empty onClick function for the next button

j0world committed:
Merge branch 'main' into feature/se-43

j0world committed:
fix: readded quiz to routes

j0world committed:
Merge branch 'main' into feature/se-43

j0world committed:
Merge branch 'main' into feature/se-43

Chrissydrx commented:
Resolve Changes and create new PR
